### ISSUE TITLE
Feature: Add ability to replace the fs module.

### ISSUE THREAD

**Issue by kurtcorbett** [orourkedd/effects-as-data-node#4]
It'd be nice to be able to use a more robust fs module like [fs-extra](https://github.com/jprichardson/node-fs-extra) instead of a promisified vanilla fs module.

**Comment by orourkedd** [comment]
yes!  do it!  It would be great to have all these functions available

**Comment by kurtcorbett** [comment]
Was thinking about a way to make that configurable so people can opt in to the fs module that they want.

**Comment by orourkedd** [comment]
you can make the fs handler export a build function so people can import the build handler function, invoke it with their own fs module, and get back an fs handler
